Automattic/jetpack
# Tiled Gallery: Set Max Columns

Sets the max columns allowed to the number of images.
Allows columns to resize when used in landscape/portrait mode.

Fixes #[4147](https://github.com/wordpress-mobile/gutenberg-mobile/issues/4147)
Fixes #[4138](https://github.com/wordpress-mobile/gutenberg-mobile/issues/4138)

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4441

#### Changes proposed in this Pull Request:
* Sets the max columns allowed to either the number of images of the `MAX_COLUMNS` value of 20 ([defined here in the block's `constant.js` file](https://github.com/Automattic/jetpack/blob/ad8c7256858e4de2e91ade9a6053e9326aa8e022/projects/plugins/jetpack/extensions/blocks/tiled-gallery/constants.js#L4)).
* Sets the max number of displayed columns in the editor to either two (for portrait mode) or four (for landscape mode). This brings the block's behaviour inline with that of the gallery block's behaviour.
* Allows columns to resize when used in landscape/portrait mode.

#### Does this pull request change what data or activity we track or use?
N/A

#### Screenshots
https://user-images.githubusercontent.com/13263478/143116610-ed2ae79c-86a0-4925-885f-78c08fd83ca1.mp4

#### Testing instructions:
**Test 1**
* Add tiled gallery block.
* Add three images.
* Open settings and verify that you can go from 1-3 columns.
* Add another image.
* Open settings and verify that you can go to 4 columns.

**Test 2**
* Add tiled gallery block.
* Add four images.
* open settings and change number of columns to 4.
* Rotate to landscape and verify that the block shows 4 columns and expands to fit the space.
